### Year 1 Topology - Limit Checking - Overloading, Voltage limits

In [1]:
import pandas as pd

##### Loading

##### Percentage Loading in Branches 

Calculation related can be found in PSSE PAGVI Manual Chapter 6 Power system network simulations - Section 6.6.7 Activity Rate 

%I = MVA(actual)/(MVA(rated)*Vpu)

Vpu -> Bus Voltage at from end 

In [2]:
list_all = []
list_gens = [0,50,100]
list_lsc = ['LLS','RLS','HLS']
for gen in list_gens:
    for lsc in list_lsc:
        file_csv_out = 'savnw_sol_' + str(gen) +'_'+ lsc + '_l.csv'
        data_load = pd.read_csv(file_csv_out,header=1, skiprows=0)
        # Extracted data contains branch loading from the metered and non-metered ends. The data is filtered to obtain the branch loading from the metered end.
        data_fil = data_load[data_load['From Bus']!=data_load['Non Metered Bus']].reset_index(drop=True)
        data_fil['From Bus V'] = data_fil['From Bus Ext'].str.extract(r'(\d+\.\d+)') 
        data_fil['To Bus V'] = data_fil['To Bus Ext'].str.extract(r'(\d+\.\d+)')
        data_fil['Scenario']= 'Solar = ' + str(gen) + ' MW, ' + lsc.upper()
        list_all.append(data_fil)
data_all = pd.concat(list_all).reset_index(drop=True)
data_all = data_all[['Scenario','From Bus', 'From Bus Ext', 'To Bus', 'To Bus Ext', 'Ckt ID',
       'Non Metered Bus', ' Branch I', 'Branch Flow MW', 'Branch Flow MVAR',
       'Branch MVA', 'MW Loss', 'MVAR Loss', 'Rate', '%I', '%MVA',
       'From Bus V', 'To Bus V']]

##### Checking transformer loading greater than 80% 

In [3]:
transformer_loading = data_all[data_all['From Bus V']!=data_all['To Bus V']].reset_index(drop=True)
transformer_loading[transformer_loading['%MVA'] > 80.0].reset_index(drop=True)

Scenario  From Bus        From Bus Ext  To Bus  \
0    Solar = 0 MW, HLS       205  SUB230      230.00     206   
1   Solar = 50 MW, HLS       205  SUB230      230.00     206   
2  Solar = 100 MW, HLS       205  SUB230      230.00     206   

           To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
0  URBGEN      18.000       1              206       2478         -875.02   
1  URBGEN      18.000       1              206       2452         -873.96   
2  URBGEN      18.000       1              206       2432         -872.94   

   Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss    Rate     %I   %MVA  \
0           -412.73      967.47     2.53     129.91  1150.0  85.84  84.13   
1           -390.62      957.28     2.48     127.19  1150.0  84.94  83.24   
2           -373.88      949.64     2.44     125.17  1150.0  84.26  82.58   

  From Bus V To Bus V  
0     230.00   18.000  
1     230.00   18.000  
2     230.00   18.000

##### Checking branch loading greater than 80% 

In [8]:
branch_loading = data_all[data_all['From Bus V'] == data_all['To Bus V']].reset_index(drop=True)
branch_loading[branch_loading['%I'] > 80.0].reset_index(drop=True)

Scenario  From Bus        From Bus Ext  To Bus  \
0  Solar = 0 MW, HLS       153  MID230      230.00     154   

           To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
0  DOWNTN      230.00       1              154        605          230.42   

   Branch Flow MVAR  Branch MVA  MW Loss  MVAR Loss   Rate     %I   %MVA  \
0             84.09      245.29     2.95      26.58  300.0  80.41  81.76   

  From Bus V To Bus V  
0     230.00   230.00

##### Voltage Upper Limits 

In [5]:
list_volt = []
list_gens = [0,50,100]
list_lsc = ['LLS','RLS','HLS']
for gen in list_gens:
    for lsc in list_lsc:
        file_csv_out = 'savnw_sol_' + str(gen) +'_'+ lsc + '_volt.csv'
        data_volt = pd.read_csv(file_csv_out,header=1, skiprows=0)
        data_volt['Scenario']= 'Solar = ' + str(gen) + ' MW, ' + lsc.upper()
        list_volt.append(data_volt)
data_all_volt = pd.concat(list_volt).reset_index(drop=True)
data_all_volt = data_all_volt[['Scenario','Bus Number', 'Bus Name', 'Area', 'Base Voltage', 'Bus Voltage(PU)','Bus Voltage(kV)']]

In [6]:
data_all_volt[data_all_volt['Bus Voltage(PU)']>1.05].reset_index(drop=True)

Scenario  Bus Number      Bus Name  Area  Base Voltage  \
0     Solar = 0 MW, LLS         201  HYDRO            2         500.0   
1     Solar = 0 MW, LLS        3018  CATDOG_G         5          13.8   
2     Solar = 0 MW, RLS         201  HYDRO            2         500.0   
3     Solar = 0 MW, RLS        3018  CATDOG_G         5          13.8   
4     Solar = 0 MW, HLS         201  HYDRO            2         500.0   
5    Solar = 50 MW, LLS         201  HYDRO            2         500.0   
6    Solar = 50 MW, LLS        3018  CATDOG_G         5          13.8   
7    Solar = 50 MW, RLS         201  HYDRO            2         500.0   
8    Solar = 50 MW, RLS        3018  CATDOG_G         5          13.8   
9    Solar = 50 MW, HLS         201  HYDRO            2         500.0   
10  Solar = 100 MW, LLS         201  HYDRO            2         500.0   
11  Solar = 100 MW, LLS        3018  CATDOG_G         5          13.8   
12  Solar = 100 MW, RLS         201  HYDRO            2         500.0   
13  Solar = 100 MW, RLS        3018  CATDOG_G         5          13.8   
14  Solar = 100 MW, HLS         201  HYDRO            2         500.0   

    Bus Voltage(PU)  Bus Voltage(kV)  
0             1.056          528.079  
1             1.055           14.559  
2             1.054          526.767  
3             1.051           14.510  
4             1.051          525.666  
5             1.058          528.781  
6             1.056           14.567  
7             1.055          527.517  
8             1.052           14.518  
9             1.053          526.453  
10            1.059          529.439  
11            1.056           14.573  
12            1.056          528.218  
13            1.053           14.525  
14            1.054          527.189

##### Voltage Lower Limits

In [7]:
data_all_volt[data_all_volt['Bus Voltage(PU)']<0.95]

Empty DataFrame
Columns: [Scenario, Bus Number, Bus Name, Area, Base Voltage, Bus Voltage(PU), Bus Voltage(kV)]
Index: []

#### Conclusion 

Year 1 Toplogy limit violation check for overloading and voltage was carried out for the three load scenarios. It was seen that
1. overloading of greater than 80% was reported for following branches and transformers for considered solar output and load scenarios 
    * Solar = 0 MW, HLS - Branch 153 MID230 230.00 154 DOWNTN 230.00 Ckt ID 1
    * Solar = 0 MW, HLS	- Transformer 205	SUB230 230.00	206	URBGEN 18.000
    * Solar = 50 MW, HLS	- Transformer 205	SUB230 230.00	206	URBGEN 18.000
    * Solar = 100 MW, HLS	- Transformer 205	SUB230 230.00	206	URBGEN 18.000
2. The buses violating the upper voltage limit for various scenarios are:
    * Solar = 0 MW, Scenarios = LLS, RLS
      BUS 201, 308
    * Solar = 0 MW, Scenarios = HLS
      BUS 201
    * Solar = 50 MW, Scenarios = LLS, RLS
      BUS 201, 308
    * Solar = 50 MW, Scenarios = HLS
      BUS 201
    * Solar = 100 MW, Scenarios = LLS, RLS
      BUS 201, 308
    * Solar = 100 MW, Scenarios = HLS
      BUS 201